### PR TITLE
[CAS-772] Move classes to offline

### PR DESCRIPTION
**Pull request by ogkuzmin**
### 🎯 Goal

Prepare codebase to move livedata to new module and keep offline without any livedata dependency

### 🛠 Implementation details
1) Some simple movements of classes and tests
2) Wrapper for `Events`. Copy to offline and wrap it in old livedata package
3) Wrapper for `RetryPolicy`
4) Wrapper for `ChannelData`
5) Wrapper for `QueryChannelsController::ChannelsState`
6) Wrapper for `ChannelController::MessagesState`
7) Some tests cleanups

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (CMS, cookbooks, tutorial)
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-offline/api/stream-chat-android-offline.api [hunk 1]
```diff
@@ -707,6 +652,94 @@ public final class io/getstream/chat/android/offline/querychannels/QueryChannels
 	public final fun setRecoveryNeeded (Z)V
 }
 
+public final class io/getstream/chat/android/offline/repository/database/ChatDatabase_Impl {
```
[the remark this answers is not in the document]
Comment (adasiewiczr) [reply stream-chat-android-offline/api/stream-chat-android-offline.api:707]: 🤔 it's generated in Java, but shouldn't we exclude such classes from `.api`?

**Review by adasiewiczr — APPROVED** [review]
Good job!

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChannelData.kt [hunk 2]
```diff
@@ -6,74 +6,129 @@ import io.getstream.chat.android.client.models.Member
...
-    /** when the channel was deleted */
-    var deletedAt: Date? = null,
-    /** channel member count */
-    var memberCount: Int = 0,
-    /** channel's team */
-    var team: String = "",
-    /** all the custom data provided for this channel */
-    var extraData: MutableMap<String, Any> = mutableMapOf()
-) {
+@Deprecated(
```
[the remark this answers is not in the document]
Comment (ogkuzmin) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChannelData.kt:15]: Yes, I added info for deprecated `ChannelData`. Also in order to provide working mechanism added to livedata `ChannelController` not deprecated live data with `OfflineChannelData`

File: stream-chat-android-offline/api/stream-chat-android-offline.api (hunk 1 above)
[the remark this answers is not in the document]
Comment (ogkuzmin) [reply stream-chat-android-offline/api/stream-chat-android-offline.api:710]: I'm not sure how to handle it. It was in API before. We can fix it after merge

**Pull request merged**